src/caffe/test/test_gradient_based_solver.cpp

```diff
@@ -565,7 +581,9 @@ class SGDSolverTest : public GradientBasedSolverTest<TypeParam> {
 
  protected:
   virtual void InitSolver(const SolverParameter& param) {
-    this->solver_.reset(new SGDSolver<Dtype>(param));
+    SolverParameter new_param = param;
+    new_param.set_type("SGD");
```

Comment:
Setting the type has no effect when instantiating the solver directly from its type as done here with `SGDSolver` and the others. It's only needed when making use of the solver registry.